### PR TITLE
[AKS] Add custom download url support for 'az aks install-cli'

### PR DESCRIPTION
**Pull request by karataliu**
**Description**  
Allow customers with no internet access to customize aks related tool download source path to an internal url.

**Testing Guide**  
az aks install-cli --base-src-url <url>

**History Notes**  
[AKS] az aks install-cli: Support customize download url

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -323,21 +323,23 @@ def dcos_install_cli(cmd, install_location=None, client_version='1.8'):
...
     """
     Install kubectl, a command-line interface for Kubernetes clusters.
     """
 
-    source_url = "https://storage.googleapis.com/kubernetes-release/release"
-    cloud_name = cmd.cli_ctx.cloud.name
-    if cloud_name.lower() == 'azurechinacloud':
-        source_url = 'https://mirror.azure.cn/kubernetes/kubectl'
+    if not source_url:
+        source_url = "https://storage.googleapis.com/kubernetes-release/release"
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:339]: how does this work in air-gapped clouds?

**Comment by yungezz** [comment]
hi @fengzhou-msft could you pls help to review this? thanks

**Review by arrownj — APPROVED** [review]
LGTM

**Review by yungezz — APPROVED** [review]

**Comment by bingosummer** [comment]
LGTM

**Pull request merged**